src/dispatch/plugins/dispatch_slack/modals.py

```diff
@@ -660,3 +683,234 @@ def add_timeline_event_from_submitted_form(action: dict, db_session=None):
+        "submit": {"type": "plain_text", "text": "Run"},
+        "callback_id": RunWorkflowCallbacks.update_view,
+        "private_metadata": json.dumps({"incident_id": str(incident.id)}),
+    }
+
+    selected_option = None
+    workflow_options = []
+    for w in workflows:
+        current_option = {
+            "text": {
+                "type": "plain_text",
+                "text": w.name,
+            },
+            "value": str(w.id),
+        }
+
+        workflow_options.append(current_option)
+
+        if selected_workflow:
+            if w.id == selected_workflow.id:
+                selected_option = current_option
+
+    if selected_workflow:
+        select_block = {
+            "block_id": RunWorkflowBlockFields.workflow_select,
+            "type": "input",
+            "element": {
+                "type": "static_select",
+                "placeholder": {
+                    "type": "plain_text",
+                    "text": "Select Workflow",
+                },
+                "initial_option": selected_option,
+                "options": workflow_options,
+                "action_id": RunWorkflowBlockFields.workflow_select,
+            },
+            "label": {"type": "plain_text", "text": "Workflow"},
+        }
+    else:
+        select_block = {
+            "block_id": RunWorkflowBlockFields.workflow_select,
+            "type": "actions",
+            "elements": [
+                {
+                    "type": "static_select",
+                    "placeholder": {
+                        "type": "plain_text",
+                        "text": "Select Workflow",
+                    },
+                    "options": workflow_options,
+                }
+            ],
+        }
+
+    modal_template["blocks"].append(select_block)
+
+    return modal_template
+
+
+@background_task
+def create_run_workflow_modal(incident_id: int, command: dict = None, db_session=None):
+    """Creates a modal for running a workflow."""
+    trigger_id = command.get("trigger_id")
+
+    incident = incident_service.get(db_session=db_session, incident_id=incident_id)
+    workflows = workflow_service.get_enabled(db_session=db_session)
+
+    if workflows:
+        modal_create_template = build_workflow_blocks(incident=incident, workflows=workflows)
+
+        dispatch_slack_service.open_modal_with_user(
+            client=slack_client, trigger_id=trigger_id, modal=modal_create_template
+        )
+    else:
+        blocks = [
+            {
+                "type": "section",
+                "text": {
+                    "type": "mrkdwn",
+                    "text": "No workflows are enabled. You can enable one in the Dispatch UI at /workflows.",
+                },
+            }
+        ]
+        dispatch_slack_service.send_ephemeral_message(
+            slack_client,
+            command["channel_id"],
+            command["user_id"],
+            "No external workflow plugin.",
```

Comment:
To be consistent with text in blocks.
```suggestion
            "No workflows enabled.",
```